### PR TITLE
openssl: Add encryption and decryption of files

### PR DESCRIPTION
**Pull request by mfrw**
----

- [ ] The page (if new), does not already exist in the repo.

- [ ] The page (if new), has been added to the correct platform folder:  
      `common/` if it's common to all platforms, `linux/` if it's Linux-specific, and so on.

- [x] The page has 8 or fewer examples.

- [x] The PR is appropriately titled:  
      `<command name>: add page` for new pages, or `<command name>: <description of changes>` for pages being edited

- [x] The page follows the [contributing](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md) guidelines

### CODE REVIEW

File: pages/common/openssl.md [hunk 1]
```diff
@@ -25,3 +25,11 @@
 - Display the complete certificate chain of an HTTPS server:
 
 `openssl s_client -connect {{host}}:443 -showcerts </dev/null`
+
+- Encrypt a file with des with asciii armored output:
```
Comment (sbrl) [inline pages/common/openssl.md]: Typo: `asciii` -> `ascii`

File: pages/common/openssl.md [hunk 2]
```diff
@@ -25,3 +25,11 @@
...
+
+`openssl des  -a -salt -in {{infile-plain}} -k {{key}} -out {{outfile-enc}}`
```
Comment (sbrl) [inline pages/common/openssl.md]: Yep, there are two spaces here!

File: pages/common/openssl.md [hunk 3]
```diff
@@ -25,3 +25,11 @@
...
+
+- Decrypt an ascii armored encrypted with des:
+
+`openssl des -d -a -salt -in {{infile-enc}} -k {{key}} -out {{outfile-plain.}}`
```
Comment (sbrl) [inline pages/common/openssl.md]: There's an extra full stop at the end of `outfile-plain`. Is this intentional?

**Comment by sbrl** [comment]
Good spot, @Darexon!

Thanks for the PR! I've left a few comments for you, @mfrw :smiley_cat:

**Comment by mfrw** [comment]
@sbrl I have incorporated the changes. Sorry for the sloppy first PR.
Wont repeat it again.
cheers

**Comment by sbrl** [comment]
No worries, @mfrw! That's what the review process is for - catching the things that you miss on your own. I for one miss things all the time that show up in the reviews :P Check out PR #1329 - I think that was my first if I remember correctly?

**Comment by agnivade** [comment]
Sorry for having the unpopular opinion here.

But I am against merging this PR. I think the new options are a bit advanced and not necessary for the scope of our project.

**Comment by mfrw** [comment]
@agnivade Sure no issue, but in my opinion these were the most basic ones i could think of, if one wanted to see how to quickly encrypt/decrypt files without having to jump to stackoverflow :)

**Comment by agnivade** [comment]
There will be some options which won't be covered in tldr pages. Please keep in mind this project does not attempt to be a man page replacement. That's what the man pages are for, if someone wants to look up advanced options.

The question here is are the encrypt/decrypt options going to be used by a beginner, or is it more or a one-off use case. I believe it to be the latter, and therefore my unwillingness to merge this.

**Comment by mfrw** [comment]
@agnivade I agree completely :)

Should we close this then ?

**Comment by agnivade** [comment]
I would leave it for @sbrl to take the final call.

**Comment by sbrl** [comment]
Hrm, I see your point, @agnivade. In addition, we've hit our limit of 8 examples for this page - which is always a problem with commands like `openssl` (`curl` is another big problem command). While I do think it has its uses, I believe that it's probably going to be easier to google than, say, the certificate manipulation examples. In addition, there's also GPG / PGP for encrypting files - which is a more common approach to take when performing encryption.

In which case, I'd say that these additions, while great, don't quite work on this page then.

Thanks for the contribution though, @mfrw :smiley_cat:

**Pull request closed without merging**